src/pyhf/cli/rootio.py

```diff
@@ -0,0 +1,100 @@
+import logging
+
+import click
+import json
+import os
+import jsonpatch
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+# This is only needed for Python 2/3 compatibility
+def ensure_dirs(path):
+    try:
+        os.makedirs(path, exist_ok=True)  # lgtm [py/call/wrong-named-argument]
+    except TypeError:
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+
+@click.group(name='rootio')
+def cli():
+    pass
+
+
+@cli.command()
+@click.argument('entrypoint-xml', type=click.Path(exists=True))
+@click.option(
+    '--basedir',
+    help='The base directory for the XML files to point relative to.',
+    type=click.Path(exists=True),
+    default=os.getcwd(),
+)
+@click.option(
+    '--output-file',
+    help='The location of the output json file. If not specified, prints to screen.',
+    default=None,
+)
+@click.option('--track-progress/--hide-progress', default=True)
+def xml2json(entrypoint_xml, basedir, output_file, track_progress):
+    """ Entrypoint XML: The top-level XML file for the PDF definition. """
+    try:
+        import uproot
+
+        assert uproot
+    except ImportError:
+        log.error(
+            "xml2json requires uproot, please install pyhf using the "
+            "xmlio extra: pip install pyhf[xmlio] or install uproot "
+            "manually: pip install uproot"
```

Comment:
I would advocate that we don't tell people to install dependencies for extras manually, but let us do it.
```suggestion
```

So I'd suggest the last two lines just be
```
            "xmlio extra: pip install pyhf[xmlio]"
```